keywords/subclassof.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+module.exports = function defFunc(ajv, customConstructors, overrideBuffer = false) {
+  var CONSTRUCTORS = {
+    Object: Object,
+    Array: Array,
+    Function: Function,
+    Number: Number,
+    String: String,
+    Date: Date,
+    RegExp: RegExp,
+    Error: Error
+  };
+
+  /* istanbul ignore else */
+  if(typeof Promise !== 'undefined')
+    addCustomConstructor(CONSTRUCTORS, 'Promise', Promise);
+
+  if (arguments.length > 1) {
+    if(typeof customConstructors !== 'object') {
+      throw new Error(
+        'second argument \'customConstructors\' should be an object with <string, function> key-value pairs'
+      );
+    }
+    Object.keys(customConstructors).forEach(function(key) {
+      if(key === 'Buffer' && !overrideBuffer) {
+        throw new Error(
+          'Buffer is not a subclassable class in ES6. See https://stackoverflow.com/questions/44909648'
+        );
+      }
+      addCustomConstructor(CONSTRUCTORS, key, customConstructors[key]);
+    });
+  }
+
+  var definition = {
+    compile: function (schema) {
+      if (typeof schema == 'string') {
+        var Constructor = getConstructor(schema);
+        return function (data) {
+          return data.prototype instanceof Constructor ||
+            (data.constructor.prototype instanceof Constructor && data !== Object);
```

Comment:
`data.constructor.prototype instanceof Constructor` is probably the same as `data instanceof Constructor`, I don't think that instances should pass this keyword, only subclasses should.